handlers/src/main/java/com/nextdoor/bender/handler/kinesis/KinesisHandler.java

```diff
@@ -70,4 +78,30 @@ public void onException(Exception e) {
   public InternalEventIterator<InternalEvent> getInternalEventIterator() {
     return this.recordIterator;
   }
+
+  @Override
+  public HandlerMetadata getHandlerMetadata() {
+    HandlerMetadata metadata = new HandlerMetadata();
+
+    /**
+     * Access this.event, which is exposed to us during the run of handler() above. Use the first
+     * record from the event to populate some metadata about the run.
+     */
+    KinesisEventRecord firstRecord = event.getRecords().get(0);
+    this.source = SourceUtils.getSource(firstRecord.getEventSourceARN(), sources);
+
+    metadata.setField("partitionKey", firstRecord.getKinesis().getPartitionKey());
+    metadata.setField("sequenceNumber", firstRecord.getKinesis().getSequenceNumber());
+    metadata.setField("arrivalTime", firstRecord.getKinesis()
+        .getApproximateArrivalTimestamp().getTime());
+    metadata.setField("sourceArn", firstRecord.getEventSourceARN());
+
+    /*
+     * TODO: Figure this one out
+     */
+    //this.timestamp = internal.getEventTime();
```

Comment:
I'm not sure we need these anymore especially because this is now metadata about the invoking event.